### DISCUSSION TITLE
What are your local dev node recommended practices?

### DISCUSSION THREAD

**Discussion by burnto** [foundry-rs/foundry#1363] [Q&A] [answered]
Hi all, coming from `hardhat` and `hardhat-deploy`, I used to use `npx hardhat node` when I wanted to run a local node with my contracts deployed and ready to integrate against as a client (to write frontends, scripts, etc).

How are all you all doing this with foundry? Is a hardhat project with foundry added still the best approach here, or is there a more foundry-only approach to running a local node and quickly deploying projects?

**Comment by onbjerg** [comment, marked as answer]
Currently you have to use something external like a local node, or Hardhat's node. However we will soon ship Anvil which is our own local development node. See https://github.com/foundry-rs/foundry/pull/1037

**Reply by burnto** [reply]
Thanks, wasn’t aware of Anvil. Looking forward to it.
